IntegrationTests/UITests/UITest.cs

```diff
@@ -38,23 +39,89 @@ void HandleApplicationIdle(object sender, EventArgs e)
             {
                 await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                 await WaitForIdleAsync();
-                var dialog = Application.OpenForms.OfType<T>().Single();
+                using var form = Application.OpenForms.OfType<T>().Single();
+                Assert.True(form.Visible, $"{form.GetType().FullName} {form.Name} should be visible");
                 try
                 {
-                    await runAsync(dialog);
+                    await runTestAsync(form);
                 }
                 finally
                 {
-                    dialog.Close();
+                    form.Close();
                 }
             });
 
-            showDialog();
+            showForm();
 
             // Join the asynchronous test operation so any exceptions are rethrown on this thread
             test.Join();
         }
 
+        public static void RunDialog<T>(
+            Action<Form> showDialog,
+            Func<T, Task> runTestAsync)
+            where T : Form
+        {
+            Form mainForm = null;
```

Comment:
💡 This can be greatly simplified:

```csharp
using var mainForm = new Form { Text = $"Test {typeof(T).Name}" };
RunForm<T>(...);
mainForm.Close();
```

The `CloseAndDispose` method is unnecessary, as are all the finally blocks which call it.

Reply:
You are right. Closing the plain `Form` is not necessary.
Should we dispose/close it in the UI thread?

Reply:
Without closing the test main form, most of the tests hang: https://github.com/gitextensions/gitextensions/issues/7682#issuecomment-577926457.

Reply:
> Without closing the test main form

Ah, yes this makes sense because `Application.Run` is used. `RunForm` knows to close a dialog at the end of the test, but in the case of this implementation of `RunDialog` it is closing the wrong form so `Application.Run` never completes.